presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java

```diff
@@ -152,12 +154,28 @@ public void testCeil()
         assertFunction("ceiling(-123.0)", DOUBLE, -123.0);
         assertFunction("ceiling(123.45)", DOUBLE, 124.0);
         assertFunction("ceiling(-123.45)", DOUBLE, -123.0);
-        assertFunction("ceiling(CAST(NULL AS DOUBLE))", DOUBLE, null);
         assertFunction("ceiling(REAL '123.0')", REAL, 123.0f);
         assertFunction("ceiling(REAL '-123.0')", REAL, -123.0f);
         assertFunction("ceiling(REAL '123.45')", REAL, 124.0f);
         assertFunction("ceiling(REAL '-123.45')", REAL, -123.0f);
+
+        assertFunction("ceiling(DECIMAL '123')", createDecimalType(3), SqlDecimal.of("123"));
```

Comment:
Split these tests into 3 sections (separated by empty lines), each covering one of short-short, long-short, long-long.